### PR TITLE
Fix errors in vis.py

### PR DESCRIPTION
**Pull request by aabobakr**
Fixing error: 
TypeError: Layout of the output array img is incompatible with cv::Mat (step[ndims-1] != elemsize or step[1] != elemsize*nchannels)

in vis_class(..) and vis_bbox(..) methods when calling vis_one_image_opencv(..).

### CODE REVIEW

**Comment by gadcam** [comment]
@aabobakr when do you have this error ? Could you share a bit more information on how to reproduce the bug this PR aims to solve ?

**Comment by aabobakr** [comment]
@gadcam I got this error when I tried to get a numpy array of with the detections (bbox and classes) visualised by calling the method:
 `im = vis_utils.vis_one_image_opencv(im[:, :, ::-1], 
            cls_boxes,
            cls_segms,
            cls_keyps,
            thresh=0.7,
            kp_thresh=2, 
            show_box=True,
            show_class=True, 
            dataset=dummy_coco_dataset
            )` 

instead of calling `vis_utils.vis_one_image(..)` in the infer_simply.py tool and this is the complete error msg: 

`Traceback (most recent call last):
  File "tools/infer_simple.py", line 181, in <module>
    main(args)
  File "tools/infer_simple.py", line 173, in main
    dataset=dummy_coco_dataset
  File "/media/dl/detectron/detectron/utils/vis.py", line 236, in vis_one_image_opencv
    im = vis_class(im, (bbox[0], bbox[1] - 2), class_str)
  File "/media/dl/detectron/detectron/utils/vis.py", line 124, in vis_class
    cv2.rectangle(img, back_tl, back_br, _GREEN, -1)
TypeError: Layout of the output array img is incompatible with cv::Mat (step[ndims-1] != elemsize or step[1] != elemsize*nchannels)
`

File: detectron/utils/vis.py [hunk 1]
```diff
@@ -113,6 +113,8 @@ def vis_mask(img, mask, col, alpha=0.4, show_border=True, border_thick=1):
 
 def vis_class(img, pos, class_str, font_scale=0.35):
     """Visualizes the class."""
+
```
Comment (ir413) [inline detectron/utils/vis.py]: Could you please remove the new line as it seems unrelated to the bug fix?

File: detectron/utils/vis.py [hunk 2]
```diff
@@ -130,6 +132,8 @@ def vis_class(img, pos, class_str, font_scale=0.35):
 
 def vis_bbox(img, bbox, thick=1):
     """Visualizes a bounding box."""
+
```
Comment (ir413) [inline detectron/utils/vis.py]: Same comment as above.

File: detectron/utils/vis.py [hunk 3]
```diff
@@ -96,7 +96,7 @@ def get_class_string(class_index, score, dataset):
 
 def vis_mask(img, mask, col, alpha=0.4, show_border=True, border_thick=1):
     """Visualizes a single binary mask."""
-
+
```
Comment (ir413) [inline detectron/utils/vis.py]: Could you remove the trailing spaces please?

File: detectron/utils/vis.py [hunk 4]
```diff
@@ -202,7 +204,7 @@ def vis_one_image_opencv(
         im, boxes, segms=None, keypoints=None, thresh=0.9, kp_thresh=2,
         show_box=False, dataset=None, show_class=False):
     """Constructs a numpy array with the detections visualized."""
-
+
```
Comment (ir413) [inline detectron/utils/vis.py]: Same comment as above.

File: detectron/utils/vis.py [hunk 5]
```diff
@@ -129,7 +129,8 @@ def vis_class(img, pos, class_str, font_scale=0.35):
 
 
 def vis_bbox(img, bbox, thick=1):
-    """Visualizes a bounding box."""
+    """Visualizes a bounding box."""
```
Comment (ir413) [inline detectron/utils/vis.py]: I think there is one more trailing space left here. Could you please delete it too? Sorry I missed it earlier.

File: detectron/utils/vis.py [hunk 6]
```diff
@@ -96,7 +96,6 @@ def get_class_string(class_index, score, dataset):
 
 def vis_mask(img, mask, col, alpha=0.4, show_border=True, border_thick=1):
     """Visualizes a single binary mask."""
-
```
Comment (ir413) [inline detectron/utils/vis.py]: It would be nice to leave this as it was (new line without trailing spaces) as the change is unrelated to the fix.

File: detectron/utils/vis.py [hunk 7]
```diff
@@ -202,7 +203,6 @@ def vis_one_image_opencv(
         im, boxes, segms=None, keypoints=None, thresh=0.9, kp_thresh=2,
         show_box=False, dataset=None, show_class=False):
     """Constructs a numpy array with the detections visualized."""
-
```
Comment (ir413) [inline detectron/utils/vis.py]: Same comment about new line as above.

File: detectron/utils/vis.py (hunk 3 above)
Comment (ir413) [inline detectron/utils/vis.py]: There are trailing spaces here again. Is your editor adding these automatically by any chance? You may want to configure your editor so that the trailing spaces are removed on save automatically (or at least highlighted as they are easy to miss otherwise).

File: detectron/utils/vis.py (hunk 4 above)
Comment (ir413) [inline detectron/utils/vis.py]: Same comment as above.

File: detectron/utils/vis.py (hunk 3 above)
Comment (aabobakr) [reply detectron/utils/vis.py]: sorry I was editing the file in the browser, so it was easy to miss these trailing spaces. It should be removed now.
